### DISCUSSION TITLE
Logging in to the log file using a command will display the password

### DISCUSSION THREAD

**Discussion by ws333666** [dbeaver/dbeaver#20288] [DBeaver Development]
### Description

eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=zh_CN
Framework arguments:  -con driver=mysql|name=test-1.1|host=xxx|port=3306|user=xxx|password=xxx|connect=true
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -data C:\DBeaver -con driver=mysql|name=xxx|host=xxx|port=3306|user=xxx|password=xxx|connect=true

!ENTRY org.eclipse.ui 2 2 2023-05-19 09:42:39.404
!MESSAGE Invalid preference category path: org.eclipse.debug.ui.ConsolePreferencePage (bundle: org.eclipse.ui.console, page: org.eclipse.ui.internal.console.ansi.preferences.AnsiConsolePreferencePage)

### DBeaver Version

DBeaver Community 23.0.4

### Operating System

windows server 2019

### Database and driver

Mysql8:Driver for MySQL 8 and later

### Steps to reproduce

1、Connect to the database using the command
2、open log file
You can see the plaintext password
or
1、Connect to the database using the command
2、open help
3、open Installation Information
4、open configuration
You can see the plaintext password

**Comment by ShadelessFox** [comment]
Hello @ws333666,

You can add the following option in the `dbeaver.ini` file after `-vmargs`:

```
-Declipse.log.include.commandline=false
```

Please let me know if your issue is solved.

**Comment by ws333666** [comment]
> 你好@ws333666,
> 
> `dbeaver.ini`您可以在文件中添加以下选项`-vmargs`：
> 
> ```
> -Declipse.log.include.commandline=false
> ```
> 
> 如果您的问题已解决，请告诉我。

ok，I modified the dbeaver.ini file, the log file does not record the password, but it still shows in the installation information.

**Comment by ws333666** [comment]
> > 你好@ws333666,
> > `dbeaver.ini`您可以在文件中添加以下选项`-vmargs`：
> > ```
> > -Declipse.log.include.commandline=false
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 如果您的问题已经解决，请告诉我。
> 
> ok，我修改了dbeaver.ini文件，日志文件没有记录密码，但是在安装信息里还是有显示的。


*** 日期：2023年6月12日星期一 中国标准时间 上午11:13:28

*** 平台详细信息：

*** 系统属性：
DRIVERS_LOCATION=C:\sso\SsoTool\drivers
applicationXMI=org.eclipse.ui.workbench/LegacyIDE.e4xmi
dbeaver.distribution.type=exe
ds.delayed.keepInstances=true
ds.delayed.keepInstances.default=true
eclipse.application=org.jkiss.dbeaver.ui.app.standalone.standalone
eclipse.commands=-os
win32
-ws
win32
-arch
x86_64
-showsplash
-launcher
C:\Program Files\DBeaver\dbeaver.exe
-name
Dbeaver
--launcher.library
C:\Program Files\DBeaver\plugins\org.eclipse.equinox.launcher.win32.win32.x86_64_1.2.700.v20221108-1024\eclipse_11801.dll
-startup
C:\Program Files\DBeaver\plugins\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar
--launcher.overrideVmargs
-data
C:\sso\SsoTool\DBeaver
-con
driver=mysql|name=xxx|host=xxx|port=3306|user=xxx|password=xxx|connect=true

**Comment by ShadelessFox** [comment]
Installation information can assist us to locate the problem a user might have. I don't see any issues with that.

**Comment by ws333666** [comment]
> 安装信息可以帮助我们确定用户可能遇到的问题。我认为这没有任何问题。

ok，After opening DBeaver, what are the function interfaces for creating a database connection? I didn’t find it. I want to open DBeaver first and use a script to connect to the database, so that the plaintext password will not be displayed.Also, thank you for your answer, sorry for your trouble

**Comment by ShadelessFox** [comment]
Please see the wiki: https://github.com/dbeaver/dbeaver/wiki/Create-Connection

Let me know if it helps.

**Comment by ws333666** [comment]
> 请查看维基： https: [//github.com/dbeaver/dbeaver/wiki/Create-Connection](https://github.com/dbeaver/dbeaver/wiki/Create-Connection)
> 
> 让我知道是否有帮助。

I want to connect through code, I want to know if there is a function interface to create a database connection, just create a database connection instead of opening the database first and then connecting
https://github.com/dbeaver/dbeaver/wiki/Command-Line

**Comment by ShadelessFox** [comment]
I'm a bit confused. Are you talking about creating a new database connection through the command line interface or programmatically in Java? In the first case, this can be achieved using the `-con` option, just like you did.

**Comment by ws333666** [comment]
> 我有点困惑。您是在谈论通过命令行界面还是在 Java 中以编程方式创建新的数据库连接？`-con`在第一种情况下，这可以像您一样使用选项来实现。

Java or C, I want to know the function interface to create a database connection

**Comment by ShadelessFox** [comment]
Using `DataSourceRegistry#addDataSource`. Please look at the code to find examples of its usage.

If you only have questions about development, I'll convert this issue into a discussion.
